### PR TITLE
[new feature]Add runtime parse dependencies support

### PR DESCRIPTION
**Pull request by Zenser**
Hey, I want add a runtime read and parse dependencies.
A component-description.json file have tags and attributes prop.
When we use thrid-party libraries, we can get better development experience.

I have write a jsdoc parse to component-description.json file, but have not published.

### CODE REVIEW

**Comment by Zenser** [comment]
@octref  Hey, bro.
I think it will resolve the problem which is every component  need make a PR to let their components support auto complete. Thanks.

File: server/src/modes/template/tagProviders/externalTagProviders.ts [hunk 1]
```diff
@@ -25,14 +25,18 @@ export const bootstrapTagProvider = getExternalTagProvider('bootstrap', bootstra
...
+export function getRuntimeTagProvider(workspacePath: string, pkg: any, dep: string): IHTMLTagProvider | null {
+  const base = 'node_modules/' + dep;
+  if (!pkg.vetur) {
+    return null;
+  }
+
   const tagsPath = ts.findConfigFile(workspacePath, ts.sys.fileExists, join(base, pkg.vetur.tags));
   const attrsPath = ts.findConfigFile(workspacePath, ts.sys.fileExists, join(base, pkg.vetur.attributes));
-
+
```
Comment (octref) [inline server/src/modes/template/tagProviders/externalTagProviders.ts]: space

File: server/src/modes/template/tagProviders/externalTagProviders.ts [hunk 2]
```diff
@@ -25,14 +25,18 @@ export const bootstrapTagProvider = getExternalTagProvider('bootstrap', bootstra
 export const buefyTagProvider = getExternalTagProvider('buefy', buefyTags, buefyAttributes);
 export const vuetifyTagProvider = getExternalTagProvider('vuetify', vuetifyTags, vuetifyAttributes);
 
-export function getQuasarTagProvider(workspacePath: string, pkg: any): IHTMLTagProvider | null {
-  const base = 'node_modules/quasar-framework';
+export function getRuntimeTagProvider(workspacePath: string, pkg: any, dep: string): IHTMLTagProvider | null {
```
Comment (octref) [inline server/src/modes/template/tagProviders/externalTagProviders.ts]: You don't need dep, `pkg.name` will do.

**Comment by octref** [comment]
Hmm, `git rebase origin/master -s ours` will actually erase all my work...I just wanted to fix the `yarn.lock` conflict.

Let me do it again.

**Comment by octref** [comment]
Commited with #980.

**Pull request closed without merging**